dataprofiler/data_readers/json_data.py

```diff
@@ -51,83 +56,201 @@ def __init__(self, input_file_path=None, data=None, options=None):
+                    found_payload_key = payload_key
+                    break
+                        
+            # Get the non-payload data
+            flattened_json = []
+            for key in json_lines:
+                if key != found_payload_key:
+                    flattened_json = flattened_json + self._find_data(json_lines[key], path=key)
+
+            # Coalesce the data together
+            json_lines = self._coalesce_dicts(flattened_json)
+
         data, original_df_dtypes = data_utils.json_to_dataframe(
             json_lines=json_lines,
             selected_columns=self.selected_keys,
             read_in_string=False
         )
-        return data, original_df_dtypes
+        self._original_df_dtypes = original_df_dtypes
+
+        if payload_data is not None:
+            self._metadata = data
+            data = payload_data
+
+        return data
 
     def _load_data_from_str(self, data_as_str):
         """
         Loads the data from a string.
-        
+
         :param data_as_str: data in string format.
         :type data_as_str: str
         :return:
         """
         try:
-            json_lines = json.loads(data_as_str)
-            data, original_df_dtypes = \
-                self._load_data_from_json_format(json_lines)
+            data = json.loads(data_as_str)
         except json.JSONDecodeError:
-            data_generator = data_utils.data_generator(data_as_str.splitlines())
-            data, original_df_dtypes = data_utils.read_json_df(
-                data_generator=data_generator,
+            data = data_utils.data_generator(data_as_str.splitlines())
+            data = data_utils.read_json(
+                data_generator=data,
                 selected_columns=self.selected_keys,
                 read_in_string=False
             )
-        self._original_df_dtypes = original_df_dtypes
         return data
 
     def _load_data_from_file(self, input_file_path):
         """
         Loads the data from a file.
-        
+
         :param input_file_path: file path to file being loaded.
         :type input_file_path: str
         :return:
         """
         self._file_encoding = data_utils.detect_file_encoding(input_file_path)
         with open(input_file_path, encoding=self.file_encoding) as input_file:
             try:
-                json_lines = json.load(input_file)
-                data, original_df_dtypes = \
-                    self._load_data_from_json_format(json_lines)
+                data = json.load(input_file)
             except (json.JSONDecodeError, UnicodeDecodeError):
                 input_file.seek(0)
-                data_generator = data_utils.generator_on_file(input_file)
-                data, original_df_dtypes = data_utils.read_json_df(
-                    data_generator=data_generator,
+                data = data_utils.generator_on_file(input_file)
+                data = data_utils.read_json(
+                    data_generator=data,
```

Comment:
Couldn't this fail as well?

Reply:
Maybe, but if it fails there, it should error out